### PR TITLE
Adding dialog errors when requests fail in interactive dialogs, fix panics and add consistency to logs

### PR DESCRIPTION
**Pull request by BenCookie95**
#### Summary
Errors from the plugin were not being surfaced to dialogs so if something went wrong there was no info for the user. I also fixed some potential panics.

Example of dialog error:
Screenshot 2024-09-18 at 2 59 33 PM

#### Ticket
https://github.com/mattermost-community/mattermost-plugin-google-drive/issues/19

### CODE REVIEW

File: server/plugin/api.go [hunk 1]
```diff
@@ -626,42 +655,48 @@ func (p *Plugin) handleFileUpload(c *Context, w http.ResponseWriter, r *http.Req
...
 	}
 	defer r.Body.Close()
 
 	fileID := request.Submission["fileID"].(string)
 	fileInfo, appErr := p.API.GetFileInfo(fileID)
 	if appErr != nil {
-		p.API.LogError("unable to fetch file info", "err", err, "fileID", fileID)
-		w.WriteHeader(http.StatusInternalServerError)
+		p.API.LogError("Unable to fetch file info", "err", appErr, "fileID", fileID)
+		p.writeInteractiveDialogError(w, DialogErrorResponse{Error: "Unable to fetch file info", StatusCode: http.StatusInternalServerError})
```
Comment (fmartingr) [inline server/plugin/api.go:667]: Unsure if we should be this explicit with errors for end users or if we should just put that something went wrong and that they should contact and administrator for assistance. The admins would have the error logs for debugging. 3/5

**Review by fmartingr — APPROVED** [review]
👍 overall. Left a comment about exposing internal logic to the end users.

File: server/plugin/api.go (hunk 1 above)
Comment (BenCookie95) [reply server/plugin/api.go:667]: Yeah you're correct, I'll adjust those

Comment (BenCookie95) [reply server/plugin/api.go]: I adjusted all of those to be the same error message

File: server/plugin/api.go [hunk 2]
```diff
@@ -458,13 +491,13 @@ func (p *Plugin) handleDriveWatchNotifications(c *Context, w http.ResponseWriter
 	}
 	changeList, err := srv.Changes.List(pageToken).Fields("*").Do()
 	if err != nil {
-		p.API.LogError("Failed to fetch google drive changes", "err", err)
+		p.API.LogError("Failed to fetch Google Drive changes", "err", err)
```
Comment (wiggin77) [inline server/plugin/api.go:494]: I like the capitalization for proper names but would prefer it be consistent across all the log lines.

File: server/plugin/create.go [hunk 3]
```diff
@@ -131,14 +131,14 @@ func (p *Plugin) handleFilePermissions(userID string, fileID string, fileAccess
 	authToken, _ := p.getGoogleUserToken(userID)
 	srv, err := drive.NewService(ctx, option.WithTokenSource(conf.TokenSource(ctx, authToken)))
 	if err != nil {
-		p.API.LogError("failed to create drive client", "err", err)
+		p.API.LogError("Failed to create drive client", "err", err)
```
Comment (wiggin77) [inline server/plugin/create.go:134]: Google Drive client

File: server/plugin/notifications.go [hunk 4]
```diff
@@ -232,25 +232,25 @@ func (p *Plugin) startDriveWatchChannel(userID string) error {
...
 	if err != nil {
-		p.API.LogError("failed to get auth token", "err", err)
+		p.API.LogError("Failed to get auth token", "err", err)
 		return err
 	}
 
 	srv, err := drive.NewService(ctx, option.WithTokenSource(conf.TokenSource(ctx, authToken)))
 	if err != nil {
-		p.API.LogError("failed to create drive service client", "err", err)
+		p.API.LogError("Failed to create drive service client", "err", err)
```
Comment (wiggin77) [inline server/plugin/notifications.go:241]: Google Drive client

**Review by wiggin77 — CHANGES_REQUESTED** [review]
Can you make the case on proper names consistent?

**Comment by BenCookie95** [comment]
> Can you make the case on proper names consistent?

I think I caught everything now

**Review by wiggin77 — APPROVED** [review]

**Pull request merged**